### PR TITLE
Remove for orelse for For

### PR DESCRIPTION
**Pull request by jacqueswww**
### What I did
We don't support For .. orelse. We should not let it blindly not generate code.

### Cute Animal Picture

```
^.^
```

### CODE REVIEW

File: vyper/parser/stmt.py [hunk 1]
```diff
@@ -534,7 +534,7 @@ def parse_for(self):
                 "..` or `for i in range(start, start + rounds): ..`"
             ), self.stmt.iter)
 
-        block_scope_id = id(self.stmt.orelse)
```
Comment (charles-cooper) [inline vyper/parser/stmt.py]: i guess the original was a little weird

File: vyper/ast.py [hunk 2]
```diff
@@ -43,9 +43,9 @@ def __init__(self, **kwargs):
             if field_name in self.get_slots():
                 setattr(self, field_name, value)
             elif value:
-                raise CompilerPanic(
-                    f'Unsupported non-empty value field_name: {field_name}, '
-                    f' class: {type(self)} value: {value}'
+                raise SyntaxException(
+                    f'Unsupported non-empty value (valid in Python, but invalid in Vyper) \n'
```
Comment (charles-cooper) [inline vyper/ast.py]: I'm not convinced this is always the correct error message, but maybe it's not important.

File: vyper/parser/stmt.py (hunk 1 above)
Comment (fubuloubu) [reply vyper/parser/stmt.py]: Is there any concern here?

File: vyper/ast.py (hunk 2 above)
Comment (fubuloubu) [reply vyper/ast.py]: `SyntaxException` is probably a little clearer. There error message is fine.

**Comment by fubuloubu** [comment]
@jacqueswww this is ready for merge, right?

**Comment by jacqueswww** [comment]
Yes, it is.

File: vyper/parser/stmt.py (hunk 1 above)
Comment (jacqueswww) [reply vyper/parser/stmt.py]: @charles-cooper BTW with the new AST classes we have a node_id attribute now :) So we can replace all these hacks with using .node_id. Nice when old hacks are not necessary anymore.

**Pull request merged**